components/src/core/drawer/context/font-scale-context.ts

```diff
@@ -0,0 +1,9 @@
+import { createContext, useContext } from 'react';
+
+type FontScaleContextType = {
+    maxScaleFont: number;
+    disableFontScaling?: boolean;
+};
+
+export const FontScaleContext = createContext<FontScaleContextType>({ maxScaleFont: 1 });
+export const useFontScaleContext = (): FontScaleContextType => useContext(FontScaleContext);
```

Comment:
The approach looks good. I would just recommend some slight tweaking:

```suggestion
import { createContext, useContext } from 'react';

type FontScaleContextType = {
    maxScale?: number;
    disableScaling?: boolean;
};

const FontScaleContext = createContext<FontScaleContextType>({ maxScale: 100, disableScaling: false });

export const useFontScale = (): FontScaleContextType => useContext(FontScaleContext);

export const FontScaleProvider: React.FC<{FontScaleContextType & children?: ReactNode}> = (props) => {
    const {maxScale, disableScaling, children} = props;
    return (
        <FontScaleContext.Provider value={{maxScale, disableScaling}}>{children}</FontScaleContext.Provider>
    );
}
```

Exporting the component will allow users to use it like this (slightly more abstracted, so the user worries less about the proper syntax):

```diff
- <FontScaleContext.Provider value={{maxScale: 1.5, disableFontScaling: false}}>{/* children */}</FontScaleContext.Provider>
+ <FontScaleProvider maxScale={1.5} disableFontScaling={false}>{children}</FontScaleProvider>

- const {maxFontScale, disableFontScaling} = useFontScaleContext();
+ const {maxScale, disableScaling} = useFontScale();
```

Reply:
Documentation and changelog will need to be updated as well.